protocols/mdns/src/behaviour/iface.rs

```diff
@@ -198,43 +201,52 @@ impl InterfaceState {
         params: &impl PollParameters,
     ) -> Option<(PeerId, Multiaddr, Instant)> {
         // Poll receive socket.
-        while self.recv_socket.poll_readable(cx).is_ready() {
-            match self
-                .recv_socket
-                .recv_from(&mut self.recv_buffer)
-                .now_or_never()
-            {
-                Some(Ok((len, from))) => {
+        loop {
+            match self.recv_socket.poll_read(cx, &mut self.recv_buffer) {
+                Poll::Ready(Ok((len, from))) => {
                     if let Some(packet) = MdnsPacket::new_from_bytes(&self.recv_buffer[..len], from)
                     {
                         self.inject_mdns_packet(packet, params);
                     }
                 }
-                Some(Err(err)) => log::error!("Failed reading datagram: {}", err),
-                None => {}
+                Poll::Ready(Err(err)) if err.kind() == std::io::ErrorKind::WouldBlock => {
+                    // Not more bytes available on the socket to read
```

Comment:
```suggestion
                    // No more bytes available on the socket to read.
```

Not a native speaker. Might be wrong.